src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt

```diff
@@ -38,9 +39,8 @@ class RestrictedPatientsService(
   fun dischargeToHospital(dischargeToHospital: DischargeToHospitalRequest): RestrictedPatientDto {
     checkNotExistingPatient(dischargeToHospital.offenderNo)
 
-    val prisonerSearchResponse = prisonerSearchApiGateway.searchByPrisonNumber(dischargeToHospital.offenderNo)
-
-    prisonerSearchResponse.firstOrNull()
+    prisonApiGateway.getOffenderBooking(dischargeToHospital.offenderNo)
+      ?.takeIf { it.inOutStatus == InOutStatus.IN }
       ?: throw NoResultsReturnedException("No prisoner search results returned for ${dischargeToHospital.offenderNo}")
```

Comment:
The main change - check the offender's status via prison-api and not prisoner offender search in case POS has not been updated yet.